### PR TITLE
contrib/example-backend: remove cluster-admin permission from service account

### PR DESCRIPTION
**Pull request by s-urbaniak**

### CODE REVIEW

File: contrib/example-backend/controllers/clusterbinding/clusterbinding_reconcile.go [hunk 1]
```diff
@@ -217,3 +224,43 @@ func (r *reconciler) ensureRBACClusterRoleBinding(ctx context.Context, clusterBi
...
+
+func (r *reconciler) ensureRBACRoleBinding(ctx context.Context, clusterBinding *kubebindv1alpha1.ClusterBinding) error {
+	binding, err := r.getRoleBinding("kube-binder", clusterBinding.Namespace)
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("failed to get RoleBinding \"kube-bind\": %w", err)
+	}
+
+	expected := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "kube-binder",
```
Comment (sttts) [inline contrib/example-backend/controllers/clusterbinding/clusterbinding_reconcile.go:236]: does this equal the service-account name?

Comment (s-urbaniak) [reply contrib/example-backend/controllers/clusterbinding/clusterbinding_reconcile.go:236]: nack, it references the kubeconfigsecretref.name with the name of the SA.

Do we want to have it the same as well? sgtm actually

File: contrib/example-backend/deploy/01-clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+- apiGroups:
+    - "kube-bind.io"
+  resources:
+    - "apiserviceexports/status"
+  verbs: ["get","patch","update"]
+- apiGroups:
+    - "kube-bind.io"
+  resources:
+    - "apiservicenamespaces"
+  verbs: ["get", "watch", "list"]
```
Comment (sttts) [inline contrib/example-backend/deploy/01-clusterrole.yaml:40]: also write

Comment (sttts) [reply contrib/example-backend/deploy/01-clusterrole.yaml:40]: oops, this is here twice.

Comment (s-urbaniak) [reply contrib/example-backend/deploy/01-clusterrole.yaml:40]: fixed

File: contrib/example-backend/deploy/bootstrap.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+	"github.com/kube-bind/kube-bind/pkg/bootstrap"
+)
+
+//go:embed *.yaml
+var raw embed.FS
+
+func Bootstrap(ctx context.Context, discoveryClient discovery.DiscoveryInterface, dynamicClient dynamic.Interface, batteriesIncluded sets.String) error {
+	return bootstrap.Bootstrap(ctx, discoveryClient, dynamicClient, batteriesIncluded, raw)
+}
```
Comment (sttts) [inline contrib/example-backend/deploy/bootstrap.go:35]: intentional here?

Comment (sttts) [reply contrib/example-backend/deploy/bootstrap.go:35]: nvmd. It is.

Comment (s-urbaniak) [reply contrib/example-backend/deploy/bootstrap.go:35]: yes, we want to bootstrap the cluster-role so it is known, it is static. it then will be referenced as a role-binding during reconcilation.

**Review by sttts — APPROVED** [review]

**Pull request merged**